config/webpack.dev.js

```diff
@@ -1,23 +1,33 @@
 const merge = require('webpack-merge');
 const webpack = require('webpack');
 const config = require('./webpack.base');
-const GLOBALS = {
-    'process.env': {
-        NODE_ENV: JSON.stringify('development'),
-    },
-};
 
 module.exports = merge.smart(config, {
+    mode: 'development',
+    devtool: 'cheap-module-source-map',
     cache: true,
-    devtool: 'inline-source-map',
     entry: {
         client: ['webpack-hot-middleware/client', 'client'],
     },
     plugins: [
-        new webpack.DefinePlugin(GLOBALS),
+        new webpack.DefinePlugin({
+            'process.env.NODE_ENV': JSON.stringify(
+                process.env.NODE_ENV || 'development'
+            ),
+        }),
         new webpack.HotModuleReplacementPlugin(),
     ],
     module: {
         rules: [],
     },
+    node: {
+        dgram: 'empty',
+        fs: 'empty',
+        net: 'empty',
+        tls: 'empty',
+        child_process: 'empty',
+    },
+    performance: {
+        hints: false,
```

Comment:
I havent run into one yet but how loud are these warnings? Seems like something we want enabled in dev so webpack will notify us of things like large bundles

Reply:
It warms mostly on file sizes everytime the bundle updates, it's annoying in development. I think it's something we must have while bundling for production but not everytime you save a file.

Reply:
IMO I think this is an important warning as the biggest benefit of Webpack is code splitting. These should be annoying to let us know while deving we should consider splitting a bundle